diff --git a/nemo_rl/models/policy/utils.py b/nemo_rl/models/policy/utils.py
index 46bbccd28..46211aaee 100644
--- a/nemo_rl/models/policy/utils.py
+++ b/nemo_rl/models/policy/utils.py
@@ -62,6 +62,8 @@ AUTOMODEL_FACTORY: Dict[str, Any] = {
     "qwen2_5_vl": AutoModelForImageTextToText,
     "qwen2_vl": AutoModelForImageTextToText,
     "qwen2_5_omni": AutoModelForTextToWaveform,
+    "qwen3_vl": AutoModelForImageTextToText,
+    "qwen3_vl_moe": AutoModelForImageTextToText,
     "qwen3_5": AutoModelForImageTextToText,
     "llava": AutoModelForImageTextToText,
     "internvl": AutoModelForImageTextToText,
@@ -78,6 +80,8 @@ if NEMO_AUTOMODEL_AVAILABLE:
         "qwen2_5_vl": NeMoAutoModelForImageTextToText,
         "qwen2_vl": NeMoAutoModelForImageTextToText,
         "qwen2_5_omni": NeMoAutoModelForTextToWaveform,
+        "qwen3_vl": NeMoAutoModelForImageTextToText,
+        "qwen3_vl_moe": NeMoAutoModelForImageTextToText,
         "qwen3_5": NeMoAutoModelForImageTextToText,
         "llava": NeMoAutoModelForImageTextToText,
         "internvl": NeMoAutoModelForImageTextToText,
